test/TerraformOutputOnlyPluginTest.groovy

```diff
@@ -0,0 +1,142 @@
+import static org.hamcrest.Matchers.containsString
+import static org.hamcrest.Matchers.hasItem
+import static org.hamcrest.Matchers.instanceOf
+import static org.hamcrest.Matchers.not
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.times;
+
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(ResetStaticStateExtension.class)
+class TerraformOutputOnlyPluginTest {
+    @Nested
+    public class Init {
+        @Test
+        void modifiesTerraformEnvironmentStage() {
+            TerraformOutputOnlyPlugin.init()
+
+            Collection actualPlugins = TerraformEnvironmentStage.getPlugins()
+            assertThat(actualPlugins, hasItem(instanceOf(TerraformOutputOnlyPlugin.class)))
+        }
+
+        @Test
+        void addsParameters() {
+            TerraformOutputOnlyPlugin.init()
+
+            def parametersPlugin = new BuildWithParametersPlugin()
+            Collection actualParms = parametersPlugin.getBuildParameters()
+
+            assertThat(actualParms, hasItem([
+                $class: 'hudson.model.BooleanParameterDefinition',
+                name: "SHOW_OUTPUTS_ONLY",
+                defaultValue: false,
+                description: "Only run 'terraform output' to show outputs, skipping plan and apply."
+            ]))
+
+            assertThat(actualParms, hasItem([
+                $class: 'hudson.model.BooleanParameterDefinition',
+                name: "JSON_FORMAT_OUTPUTS",
+                defaultValue: false,
+                description: "Render 'terraform output' results as JSON. Only applies if SHOW_OUTPUTS_ONLY is selected."
+            ]))
+
+            assertThat(actualParms, hasItem([
+                $class: 'hudson.model.StringParameterDefinition',
+                name: "REDIRECT_OUTPUTS_TO_FILE",
+                defaultValue: "",
+                description: "Filename relative to the current workspace to redirect output to. Only applies if 'SHOW_OUTPUTS_ONLY' is selected."
+            ]))
```

Comment:
I might consider breaking this test case up into 3 separate test cases, and asserting on each parameter individually.  But that's 100% optional.

Reply:
This makes sense to me. I'll update the PR with this change.